pyseir/models/nowcast_seir_model.py

```diff
@@ -0,0 +1,942 @@
+    """
+
+    FRACTION_HOSP_DWELL_AS_DELAY = 1.0
+
+    def __init__(
+        self,
+        # ____________These are (to be) trained factors that are passed in___________
+        median_age=None,
+        # TODO simplify - not using linear ramp for either of these anywhere
+        lr_fh=(0.74, 0.0),  # geometric mean of all states
+        lr_fd=(0.85, 0.0),  # geometric mean of all states
+        delay_ci_h=0,  # added days of delay between infection and hospitalization
+        age_eval_delay=None,  # else auto determined from t_i and t_h
+        th_elongation_fraction=0.4,  # what fraction longer t_h is for older people
+        # smape values: 0 - > .333, .3 -> .319, .5 -> .319, .8 -> .321
+    ):
+        # Retain passed in parameters
+        self.lr_fh = lr_fh
+        self.lr_fd = lr_fd
+        self.delay_ci_h = delay_ci_h  # not using this yet
+        self.age_eval_delay = (
+            age_eval_delay  # Delay for median age from cases being applied to death transition
+        )
+
+        # __________Fixed parameters not trained_________
+        self.t_e = 2.0  # 2.0  sigma was 1/3. days^-1 below
+        # TODO serial period as ~5 days in our old model runs. How is this not 6 = 3 + 1/2 * 6?
+        self.t_i = 6.0  # delta was 1/6. days^-1 below
+        self.serial_period = self.t_e + 0.5 * self.t_i  # this is 5 days agrees with delta below
+
+        # Hospitalization stay length is important driver of total Hospitalizations (due to accumulation)
+        # Range of reported values from 5-8 days. Some examples
+        # https://www.medrxiv.org/content/medrxiv/early/2020/05/05/2020.04.30.20084780.full.pdf
+        self.th0 = 6.0
+
+        # Hospitalization dwell time is longer for old people
+        def th_f(med_age=None):
+            if med_age is None:  # not used much as median age data in future is now generated
+                return self.th0
+            else:
+                return self.th0 * (
+                    1.0 + th_elongation_fraction * max(0.0, min(1.0, (med_age - 30.0) / 30.0))
+                )
+
+        self.t_h = th_f
+
+        if self.age_eval_delay is None:
+            self.age_eval_delay = (
+                self.t_i + NowcastingSEIRModel.FRACTION_HOSP_DWELL_AS_DELAY * self.t_h()
+            )
+        self.fw0 = 0.5
+        self.pos0 = 0.5  # max positivity for testing -> 0
+        self.pos_c = 1.75
+        self.pos_x0 = 2.0
+        # Solution that ensures continuity of function below and its derivative
+        self.pos_b = (3.0 * self.pos_x0 - self.pos_c) / (4.0 * self.pos_x0 ** 1.5)
+        self.pos_d = self.pos_x0 ** 0.5 / 4.0 * (3.0 * self.pos_c - self.pos_x0)
+
+    def get_calibration(self):
+        """
+        Get the current calibration which is only using one proportionality factor for each of
+        fh, fd
+        """
+        return (self.lr_fh[0], self.lr_fd[0])
+
+    def run_stationary(self, rt, median_age, t_over_x, x_is_new_cases=True):
+        """
```

Comment:
Adding a doc string here with more context and the inputs and outputs would be great

Reply:
Added docstring here and in many other places